### PR TITLE
Improve documentation for `float` and `int` types

### PR DESCRIPTION
**Pull request by MDLC01**
This small documentation PR improves the documentation for the `float` and `int` types.

### CODE REVIEW

File: crates/typst/src/foundations/float.rs [hunk 1]
```diff
@@ -65,8 +66,8 @@ impl f64 {
 
     /// Checks if a float is infinite.
     ///
-    /// For floats, there is positive and negative infinity. This function
-    /// returns `true` if the float is either positive or negative infinity.
+    /// Floats can represent positive and negative infinity. This function
+    /// returns `{true}` if the float is either an infinity.
```
Comment (xkevio) [inline crates/typst/src/foundations/float.rs:70]: the `an` change doesn't make sense and is also incorrect grammar. leaving it as it was is fine

Comment (MDLC01) [reply crates/typst/src/foundations/float.rs:70]: I'm not a native English speaker so I may be wrong, but, from my understanding, "infinity" can be used as a noun. There are two infinities: negative infinity, and positive infinity. I can refer to one of them with "an infinity".

Comment (xkevio) [reply crates/typst/src/foundations/float.rs:70]: It depends on the context whether "infinity" is an uncountable noun, meaning it needs no article, or a countable one. Regardless, it is wrong in this context especially with the "either" as that would have to be followed by an "or" at some point.

Comment (MDLC01) [reply crates/typst/src/foundations/float.rs:70]: Oh you're right! I meant to write "the float is an infinity" without "either". Would that be correct?

Comment (laurmaedje) [reply crates/typst/src/foundations/float.rs:70]: Yeah, that'd be correct.

File: crates/typst/src/foundations/float.rs [hunk 2]
```diff
@@ -27,10 +27,11 @@ impl f64 {
     /// Converts a value to a float.
     ///
     /// - Booleans are converted to `0.0` or `1.0`.
-    /// - Integers are converted to the closest 64-bit float.
+    /// - Integers are converted to the closest 64-bit float. For integers with
+    ///   absolute value less than 9 quadrillion, this conversion is exact.
```
Comment (laurmaedje) [inline crates/typst/src/foundations/float.rs:31]: I think it would be write the exact number, but using exponential notation, i.e. 2^53.

**Comment by laurmaedje** [comment]
Thanks!

**Pull request merged**